viral/single_session.py

```diff
@@ -388,11 +387,26 @@ def summarise_trial(trial: TrialInfo, wheel_circumference: float) -> TrialSummar
             step_size=30,
             sampling_rate=30,
         )[0].speed,
+        speed_nonAZ=get_speed_positions(
+            position=position,
+            first_position=0,
+            last_position=150,
+            step_size=30,
+            sampling_rate=30,
+        )[0].speed,
+        trial_speed=get_speed_positions(
+            position=position,
+            first_position=0,
+            last_position=180,
+            step_size=30,
+            sampling_rate=30,
+        )[0].speed,
```

Comment:
I think you've incorrectly copied the logic from above. The [0] index only works for speed_AZ because the step_size = last_postition - first_position

Reply:
I changed the speed to be just a list of speed positions so we can take this information later on. I set the step size to 20 instead of 30, if that is alright?
```suggestion
        licks_AZ=get_anticipatory_licking(
            licks_to_position(trial, wheel_circumference)
        ),
        rewarded=trial.texture_rewarded,
        reward_drunk=reward_drunk(trial, wheel_circumference),
        trial_time_overall=trial.trial_end_time - trial.trial_start_time,
    )


def get_binned_licks(trials: List[TrialInfo], wheel_circumference: float) -> List[int]:
    """Bin licks across trials. Need to think about doing this for separate trials for the error"""
    licks = np.concatenate(
        [licks_to_position(trial, wheel_circumference) for trial in trials]
    )
```